packages/provider-keeper/tests/ui.spec.ts

```diff
@@ -0,0 +1,396 @@
+        smart: 'waves private node seed with smart account',
+    };
+
+    this.timeout(10 * m);
+    let tabWavesKeeper, tabUI;
+
+    before(async function () {
+        this.timeout(5 * m);
+        const ext: Promise<string> = fetchExtension(extension);
+        const local: StartedNetwork = await new Network().start();
+
+        // ui container
+        cUI = await (await GenericContainer.fromDockerfile(ui.dockerFilePath).build())
+            .withNetworkMode(local.getName())
+            .withNetworkAliases(ui.host)
+            .start();
+
+        // selenium container
+        cSelenium = await new GenericContainer('selenium/standalone-chrome')
+            .withExposedPorts(4444)
+            .withNetworkMode(local.getName())
+            .start();
+
+        // selenium webdriver
+        driver = new Builder()
+            .forBrowser('chrome')
+            .usingServer(`http://${cSelenium.getHost()}:${cSelenium.getMappedPort(4444)}/wd/hub`)
+            .setChromeOptions(new chrome.Options().addExtensions(await ext))
+            .build();
+
+        await setupWavesKeeperAccounts(extension, driver, accounts);
+
+        // prepare browse keeper and ui tabs
+        await driver.get(`chrome-extension://${extension}/popup.html`);
+        tabWavesKeeper = await driver.getWindowHandle();
+
+        await driver.switchTo().newWindow('tab');
+        await driver.get(`http://${ui.host}:${ui.port}`);
+        await driver.wait(until.elementLocated(By.id('sign-in')), timeout);
+        tabUI = await driver.getWindowHandle();
+        await driver.sleep(2000);
+    });
+
+    after(async function () {
+        this.timeout(60 * s);
+        driver && (await driver.quit());
+        cSelenium && (await cSelenium.stop());
+        cUI && (await cUI.stop());
+    });
+
+    it('auth tx', async () => {
+        await driver.switchTo().window(tabUI);
+
+        const loginBtn = await driver.wait(until.elementLocated(By.css('#sign-in:not(.disabled)')), timeout);
+        await loginBtn.click();
+
+        await driver.switchTo().window(tabWavesKeeper);
+        // site permission request
+        await driver.wait(until.elementLocated(By.xpath("//div[contains(@class, '-originAuthTx')]")));
+        let acceptBtn = await driver.findElement(By.css('.app button[type=submit]'));
+        await acceptBtn.click();
+        // site auth request
+        await driver.wait(until.elementLocated(By.xpath("//div[contains(@class, '-authTx')]")));
+        acceptBtn = await driver.findElement(By.css('.app button[type=submit]'));
+        await acceptBtn.click();
+        // close window
+        const closeBtn = await driver.wait(until.elementLocated(By.xpath("//button[contains(@class, '-closeBtn')]")));
+        await closeBtn.click();
+
+        await driver.switchTo().window(tabUI);
+        const userData = await driver.executeScript(() => {
+            return (window as any).getOutput();
+        });
+        expect(userData.signature).to.exist;
```

Comment:
Signature verification would make a lot of sence here. For example, using ts-lib-crypto method to check that the signature is valid: https://github.com/wavesplatform/ts-lib-crypto/blob/master/src/index.ts#L32

Reply:
Probably, this should be done in WavesKeeper e2e tests, and not in ProviderKeeper.